File: csharp/03 - log-levels/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine) => logLine.Split(':')[1].Trim();

    public static string LogLevel(string logLine) 
    {
        char[] charsToTrim = { '[', ']' };
        return logLine.Split(':')[0].ToLower().Trim(charsToTrim);
    } 

    public static string Reformat(string logLine) => $"{Message(logLine)} ({LogLevel(logLine)})";

File: cpp/04 - log-levels/log_levels.cpp
#include <string>

namespace log_line {

    std::string message(std::string inputlog) {
        return inputlog.substr(inputlog.find(" ")+1);
    }

    std::string log_level(std::string inputlog) {
        return inputlog.substr(1,inputlog.find("]")-1);
    }

    std::string reformat(std::string inputlog) {
        return message(inputlog) + " (" + log_level(inputlog) + ")";
    }

} // namespace log_line
